### PR TITLE
Enable flock and user_xattr Lustre options by default

### PR DESCRIPTION
**Pull request by sean-smith**
From the [lustre docs](http://wiki.lustre.org/Mounting_a_Lustre_File_System_on_Client_Nodes):

> `flock`: This has no measurable performance impact in modern (2.x) versions of Lustre (this page previously stated there was a performance impact; this is not correct for modern versions). This is the recommended mode, and is enabled by default in Lustre 2.13 and newer.

> `user_xattr`: Enable support for user extended attributes. Extended attributes are name:value pairs associated permanently with files and directories, similar to the environment strings associated with a process.

> `noatime` This option turns off inode access time updates.

For `systemd` based systems I've added two more options:

> `x-systemd.automount` adds to `fstab` so filesystem is automatically mounted after reboots.

> `x-systemd.requires=lnet.service` requires lustre kernel module

Signed-off-by: Sean Smith <seaam@amazon.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by sean-smith** [comment]
@lukeseawalker that's Centos 7, Ubuntu 16/18, and Amazon Linux 2

Centos 6 and Amazon Linux 1 are non-systemd.

File: recipes/fsx_mount.rb [hunk 1]
```diff
@@ -29,13 +29,19 @@
     action :create
   end
 
+  mount_options = %w[defaults _netdev flock user_xattr noauto]
```
Comment (lukeseawalker) [inline recipes/fsx_mount.rb]: I would follow the doc moving the noauto into the systemd case

**Comment by sean-smith** [comment]
@lukeseawalker The `noatime` is in the FSx docs: https://docs.aws.amazon.com/fsx/latest/LustreGuide/mounting-ec2-instance.html

**Comment by lukeseawalker** [comment]
@sean-smith ` noauto != noatime`

**Pull request merged**